tuxemon/states/combat/combat.py

```diff
@@ -1215,48 +1215,78 @@ def evolve(self) -> None:
         self.client.pop_state()
         for monster in self.players[0].monsters:
             for evolution in monster.evolutions:
+                evolved = Monster()
+                evolved.load_from_db(evolution.monster_slug)
                 if evolution.path == "standard":
                     if evolution.at_level <= monster.level:
-                        tools.open_dialog(
-                            local_session,
-                            [
-                                T.format(
-                                    "evolution_confirmation",
-                                    {"name": monster.name},
-                                )
-                            ],
-                        )
-                        tools.open_choice_dialog(
-                            local_session,
-                            menu=(
-                                (
-                                    "yes",
-                                    T.translate("yes"),
-                                    partial(self.positive_answer, monster),
-                                ),
-                                (
-                                    "no",
-                                    T.translate("no"),
-                                    self.negative_answer,
-                                ),
-                            ),
-                        )
+                        self.question_evolution(monster, evolved)
+                elif evolution.path == "gender":
+                    if evolution.at_level <= monster.level:
+                        if evolution.gender == monster.gender:
+                            self.question_evolution(monster, evolved)
+                elif evolution.path == "element":
+                    if evolution.at_level <= monster.level:
+                        if self.players[0].has_type(evolution.element):
```

Comment:
just confirming how this works...  if a monster is ready to evolve along the "element" path, this check looks at all the monsters in the party?  so the requirement to evolve here is that *any* monster in the party will enable it, not the monster itself?  im a bit fuzzy here, just trying to understand.

Reply:
same question for the "tech" evolution below.

Reply:
Exactly, @Sanglorian confirmed that it was related to any monster in the party as well as that there is no minimum level of evolution.
So if you have a Vivipere in your party (any level) and after a battle you have a monster wood type (element) or a monster with flamethrower (tech) or you're inside (location), Vivipere evolves.

Reply:
ok.  so this is expected behavior?

Reply:
Yes